pkg/controller/scaledobject/scaledobject_controller.go

```diff
@@ -355,6 +356,12 @@ func (r *ReconcileScaledObject) scaledObjectGenerationChanged(logger logr.Logger
 func (r *ReconcileScaledObject) newHPAForScaledObject(logger logr.Logger, scaledObject *kedav1alpha1.ScaledObject) (*autoscalingv2beta1.HorizontalPodAutoscaler, error) {
 	deploymentName := scaledObject.Spec.ScaleTargetRef.DeploymentName
 	scaledObjectMetricSpecs, err := r.getScaledObjectMetricSpecs(logger, scaledObject, deploymentName)
+	labels := map[string]string{
+		"app.kubernetes.io/name": getHpaName(deploymentName),
+		"app.kubernetes.io/version": version.Version,
+		"app.kubernetes.io/part-of": deploymentName,
```

Comment:
I think that we should reference `ScaledObject` instead of Deployment in here. WDYT?

Reply:
Yes, I think that is better. The jobs also reference the scaledobject.